raptor_dbw_can/src/DbwNode.cpp

```diff
@@ -848,9 +851,46 @@ void DbwNode::recvCAN(const can_msgs::msg::Frame::SharedPtr msg) {
                 }
             } break;
         }
+
+        // generates TireTemp messages and assign them to TireTempReport.
+        tire_temp_report_msg.front_left = DbwNode::generateTireTemp(fl_tire_temperature);
+        tire_temp_report_msg.front_right = DbwNode::generateTireTemp(fr_tire_temperature);
+        tire_temp_report_msg.rear_left = DbwNode::generateTireTemp(rl_tire_temperature);
+        tire_temp_report_msg.rear_right = DbwNode::generateTireTemp(rr_tire_temperature);
     }
 }
 
+// computes values of TireTemp message for a given array of tire temperatures.
+deep_orange_msgs::msg::TireTemp DbwNode::generateTireTemp(float32[] tire_temperature) {
+    deep_orange_msgs::msg::TireTemp tireTemp;
+    float32 sum = 0.0;
+    int count = 0;
+    float32[] temps;
```

Comment:
I would prefer that we use a `std::vector` or `std::list` here instead. 

also for efficiency sake, we could make a new class member that can be used here. so basically you allocate a single list/vector once and re-use it every time you call this function. This way you're not allocating and deallocating memory every time you call this function

Reply:
Minor improvement, but worth doing if doesn't take much of your time